src/marqo/core/semi_structured_vespa_index/semi_structured_document.py

```diff
@@ -76,10 +76,11 @@ def extract_field(cls, fields, name: str, default: Any):
 
     @classmethod
     def from_marqo_document(cls, document: Dict, marqo_index: SemiStructuredMarqoIndex) -> "SemiStructuredVespaDocument":
-        """Instantiate an UnstructuredVespaDocument from a valid Marqo document for feeding to Vespa"""
+        """Instantiate an SemiStructuredVespaDocument from a valid Marqo document for feeding to Vespa"""
 
         if index_constants.MARQO_DOC_ID not in document:
-            raise VespaDocumentParsingError(
+            # Please note we still use unstructured in the error message since it will be exposed to user
+            raise MarqoDocumentParsingError(
```

Comment:
Reason of the change: We are converting Marqo doc to Vespa doc, so we should raise MarqoDocumentParsingError instead of VespaDocumentParsingError